dataprofiler/tests/labelers/test_data_labelers.py

```diff
@@ -36,47 +36,79 @@ def test_train_method_exists(self):
+
+        with self.assertRaisesRegex(TypeError,
+                                    "Default label must be a string."):
+            dp.train_structured_labeler(
+                pd.DataFrame([]), default_label=1, save_dirpath=None)
 
         # doesn't accept text data
         text_data = dp.Data(data='test', data_type='text')
         with self.assertRaisesRegex(TypeError,
                                     "Input data must be either a "
                                     "`pd.DataFrame` or a `data_profiler.Data` "
                                     "and not of type `TextData`."):
-            dp.train_structured_labeler(text_data, None)
+            dp.train_structured_labeler(
+                text_data, default_label=None, save_dirpath=None)
 
         with self.assertRaisesRegex(ValueError,
                                     "The `save_dirpath` is not valid or not "
                                     "accessible."):
-            dp.train_structured_labeler(pd.DataFrame([]), "/a/test")
+            dp.train_structured_labeler(
+                pd.DataFrame([]), default_label=None, save_dirpath="/a/test")
 
         try:
             data = {'UNKNOWN': ["Beep", "Boop"],
                     'PERSON': ["GRANT", "MENSHENG"]}
             df = pd.DataFrame(data=data)
-            dp.train_structured_labeler(df, save_dirpath=None)
+            dp.train_structured_labeler(
+                df, default_label=None, save_dirpath=None)
 
             fake_data = dp.Data(data=df, data_type='csv')
-            dp.train_structured_labeler(fake_data, save_dirpath=None)
+            dp.train_structured_labeler(
+                fake_data, default_label=None, save_dirpath=None)
 
             fake_data = dp.Data(data=df, data_type='json')
-            dp.train_structured_labeler(fake_data, save_dirpath=None)
+            dp.train_structured_labeler(
+                fake_data, default_label=None, save_dirpath=None)
 
             fake_data = dp.Data(data=df, data_type='parquet')
-            dp.train_structured_labeler(fake_data, save_dirpath=None)
+            dp.train_structured_labeler(
+                fake_data, default_label=None, save_dirpath=None)
+
+        except Exception as e:
+            self.fail(str(e))
+
+        # test default label
+        data = {'LABEL1': ["word1", "word2"],
+                'LABEL2': ["word3", "word4"]}
+        df = pd.DataFrame(data=data)
+        
+        with self.assertRaisesRegex(ValueError,
+                                    "The `default_label` of UNKNOWN must "
+                                    "exist in the label mapping."):
+            dp.train_structured_labeler(
+                df, default_label=None, save_dirpath=None)
+
+        try:
+            default_label = 'LABEL1'
+            data_labeler = dp.train_structured_labeler(
+                df, default_label=default_label, save_dirpath=None)
+            self.assertTrue(default_label in data_labeler.label_mapping)
```

Comment:
I think we should also assert that it is set as the default label

Reply:
fair. Added